compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -11949,7 +11949,8 @@ private STNode parseQueryExprRhs(STNode queryConstructType, boolean isRhsExpr, b
             }
         }
 
-        if (peek().kind == SyntaxKind.DO_KEYWORD && (!isNestedQueryExpr() || selectClause == null)) {
+        if (peek().kind == SyntaxKind.DO_KEYWORD && 
```

Comment:
Shall we add few parser tests for this. include some with collect clause inside do, group-by clauses

Reply:
Added in commit [#6edfe50](https://github.com/ballerina-platform/ballerina-lang/pull/41413/commits/6edfe50fb322850be8689b622fe8dc2e00155b1a)